### PR TITLE
Migrate `MultisigTransaction` to `zod`

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This migrates the validation of `MultisigTransaction`s to `zod`.

## Changes

- Replace `confirmationSchema` with `ConfirmationSchema`
- Replace `multisigTransactionSchema` with `MultisigTransactionSchema`
- Remove `multisigTransactionPageSchema`
- Infer types from the schemas and propegate them.
- Add appropriate test coverage for the above

### CODE REVIEW

File: src/domain/messages/entities/message-confirmation.entity.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { MessageConfirmationSchema } from '@/domain/messages/entities/schemas/message.schema';
 import { z } from 'zod';
 
+// TODO: Move to a shared location as used for other entities
```
Comment (iamacook) [inline src/domain/messages/entities/message-confirmation.entity.ts:4]: I didn't do this to reduce the footprint of the PR.

File: src/domain/safe/transaction-type.validator.ts [hunk 2]
```diff
@@ -87,6 +91,11 @@ export class TransactionTypeValidator
 
     this.jsonSchemaService.getSchema(DATA_DECODED_SCHEMA_ID, dataDecodedSchema);
 
+    this.jsonSchemaService.getSchema(
+      CONFIRMATION_SCHEMA_ID,
+      confirmationSchema,
+    );
+
```
Comment (iamacook) [inline src/domain/safe/transaction-type.validator.ts:98]: This is temporarily needed until we migrate the `transactionTypePageSchema`.

File: src/routes/transactions/__tests__/controllers/list-multisig-transactions-by-safe.transactions.controller.spec.ts [hunk 3]
```diff
@@ -168,18 +164,14 @@ describe('List multisig transactions by Safe - Transactions Controller (Unit)',
       status: 200,
     });
     networkService.get.mockResolvedValueOnce({
-      data: { ...page, count: null },
+      data: { ...page, count: '1' },
```
Comment (iamacook) [inline src/routes/transactions/__tests__/controllers/list-multisig-transactions-by-safe.transactions.controller.spec.ts:167]: `count` is nullable according to our types, and the Transaction Service.

File: src/routes/transactions/entities/transaction-data.entity.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@ export class TransactionData {
     hexData: string | null,
     dataDecoded: DataDecoded | null,
     to: AddressInfo,
-    value: string,
+    value: string | null,
```
Comment (iamacook) [inline src/routes/transactions/entities/transaction-data.entity.ts:31]: As we weren't inferring the types from the schema, this was not caught. It matches the types of the Transaction Service.

File: src/routes/transactions/mappers/multisig-transactions/multisig-transaction-status.mapper.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ export class MultisigTransactionStatusMapper {
     }
     if (
       (transaction.confirmations?.length || 0) <
-      transaction.confirmationsRequired
+      (transaction.confirmationsRequired || 0)
```
Comment (iamacook) [inline src/routes/transactions/mappers/multisig-transactions/multisig-transaction-status.mapper.ts:22]: As before regarding matching the Transaction Service.

File: src/domain/safe/entities/schemas/multisig-transaction.schema.ts [hunk 6]
```diff
@@ -1,6 +1,60 @@
...
 
+export const ConfirmationSchema = z.object({
+  owner: AddressSchema,
+  submissionDate: z.coerce.date(),
+  transactionHash: HexSchema.nullish().default(null),
+  signatureType: z.nativeEnum(SignatureType),
+  signature: HexSchema.nullish().default(null),
+});
+
+export const MultisigTransactionSchema = z.object({
```
Comment (fmrsabino) [inline src/domain/safe/entities/schemas/multisig-transaction.schema.ts:20]: I'd merge the schemas with the respective entity declarations.

Comment (fmrsabino) [inline src/domain/safe/entities/schemas/multisig-transaction.schema.ts:20]: The `multisigTransactionPageSchema` and `MULTISIG_TRANSACTION_PAGE_SCHEMA_ID` are no longer needed right?

File: src/routes/transactions/__tests__/controllers/list-multisig-transactions-by-safe.transactions.controller.spec.ts (hunk 3 above)
Comment (fmrsabino) [reply src/routes/transactions/__tests__/controllers/list-multisig-transactions-by-safe.transactions.controller.spec.ts:167]: If `count` is nullable why did we have to change the `count` from `null` to `1` here? 🤔

File: src/routes/transactions/entities/multisig-execution-info.entity.ts [hunk 7]
```diff
@@ -8,16 +8,16 @@ import {
...
+  confirmationsRequired: number | null;
   @ApiProperty()
   confirmationsSubmitted: number;
   @ApiPropertyOptional({ type: AddressInfo, isArray: true, nullable: true })
   missingSigners: AddressInfo[] | null;
 
   constructor(
     nonce: number,
-    confirmationsRequired: number,
+    confirmationsRequired: number | null,
```
Comment (fmrsabino) [inline src/routes/transactions/entities/multisig-execution-info.entity.ts:20]: Is this field nullable? If so I think we should highlight this in the PR description / commit.

File: src/domain/safe/entities/schemas/multisig-transaction.schema.ts (hunk 6 above)
Comment (iamacook) [reply src/domain/safe/entities/schemas/multisig-transaction.schema.ts]: Merged in d8857aa928a976a49f366a3391f7ba5de4a985e8.

File: src/routes/transactions/entities/multisig-execution-info.entity.ts (hunk 7 above)
Comment (iamacook) [reply src/routes/transactions/entities/multisig-execution-info.entity.ts]: This is seemingly an issue in the _current_ schema. It's marked as nullable but it _isn't_ on the Transaction Service. I've undone these changes in d8857aa928a976a49f366a3391f7ba5de4a985e8.

**Comment by iamacook** [comment]
> If `count` is nullable why did we have to change the `count` from `null` to `1` here? 🤔 

It won't let me respond to this. It was a poor invalid value. It won't validate a string. I changed it to `"invalid"` in d8857aa928a976a49f366a3391f7ba5de4a985e8 for clarity.

**Comment by fmrsabino** [comment]
> > If `count` is nullable why did we have to change the `count` from `null` to `1` here? 🤔
> 
> It won't let me respond to this. It was a poor invalid value. It won't validate a string. I changed it to `"invalid"` in [d8857aa](https://github.com/safe-global/safe-client-gateway/commit/d8857aa928a976a49f366a3391f7ba5de4a985e8) for clarity.

Oh I see. Thanks for clarifying! Now I understand that it was the difference between a valid `number` vs a numeric string 👍 Thanks!

File: src/routes/transactions/entities/multisig-execution-info.entity.ts (hunk 7 above)
Comment (iamacook) [reply src/routes/transactions/entities/multisig-execution-info.entity.ts]: I've confirmed with Core that it's a required field.

**Pull request merged**